### PR TITLE
Add support in aws-s3 input for s3 notification from SNS to SQS

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
## What does this PR do?

This PR is to add support for S3 notification from SNS to SQS for `aws-s3` input. When parsing the SQS message body, if there is a `topicArn` field that exists, then it means the notification message is from SNS to SQS. 

Quoting from @christianherweg0807 from https://github.com/elastic/beats/pull/25492:
Screen Shot 2021-11-03 at 4 06 12 PM

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

`arn:aws:sqs:us-east-1:428152502467:test-fb-ks` is setup with SNS to SQS for testing.

## Related issues

- Closes https://github.com/elastic/beats/issues/13585

### CODE REVIEW

**Comment by aspacca** [comment]
👍 

could you add an integration test as well?

**Review by aspacca — APPROVED** [review]

**Comment by kaiyan-sheng** [comment]
@aspacca integration test added 🙂

File: x-pack/filebeat/input/awss3/input_integration_test.go [hunk 1]
```diff
@@ -440,8 +442,74 @@ func TestPaginatorListPrefix(t *testing.T) {
 }
 
 func TestGetProviderFromDomain(t *testing.T) {
```
Comment (aspacca) [inline x-pack/filebeat/input/awss3/input_integration_test.go:447]: I realised this should not be in `input_integration_test.go` :)

File: x-pack/filebeat/input/awss3/input_integration_test.go [hunk 2]
```diff
@@ -48,9 +47,11 @@ const (
 )
 
 type terraformOutputData struct {
-	AWSRegion  string `yaml:"aws_region"`
-	BucketName string `yaml:"bucket_name"`
-	QueueURL   string `yaml:"queue_url"`
+	AWSRegion   string `yaml:"aws_region"`
+	BucketName  string `yaml:"bucket_name"`
+	QueueURL    string `yaml:"queue_url"`
+	BucketName2 string `yaml:"bucket_name_for_sns"`
```
Comment (aspacca) [inline x-pack/filebeat/input/awss3/input_integration_test.go:56]: nit: `BucketNameForSNS`

File: x-pack/filebeat/input/awss3/input_integration_test.go [hunk 3]
```diff
@@ -48,9 +47,11 @@ const (
...
+	QueueURL2   string `yaml:"queue_url_for_sns"`
```
Comment (aspacca) [inline x-pack/filebeat/input/awss3/input_integration_test.go:57]: nit: `QueueURLForSNS`

**Review by aspacca — APPROVED** [review]

**Pull request merged**